Python/compile.c

```diff
@@ -567,6 +567,7 @@ PyCodeObject *
 _PyAST_Compile(mod_ty mod, PyObject *filename, PyCompilerFlags *pflags,
                int optimize, PyArena *arena)
 {
+    assert(!PyErr_Occurred());
```

Comment:
This feels like an overly generalized handler, and IIRC we should already have an ast validation phase at the parser when the debug mode is applied. Is there a case where this catches something and that doesn't (I assume manually produced trees might fit but they are always validated).

Reply:
IIUC the bug this PR is fixing is in the AST validation code. 

I don't think anyone should call ``_PyAST_Compile`` with an exception set.